mesonbuild/interpreter/interpreter.py

```diff
@@ -1620,15 +1620,24 @@ def func_subdir_done(self, node, args, kwargs):
                       'build_always', 'capture', 'depends', 'depend_files', 'depfile',
                       'build_by_default', 'build_always_stale', 'console', 'env',
                       'feed', 'install_tag'})
-    @typed_pos_args('custom_target', str)
-    def func_custom_target(self, node: mparser.FunctionNode, args: T.Tuple[str], kwargs: 'TYPE_kwargs') -> build.CustomTarget:
+    @typed_pos_args('custom_target', optargs=[str])
+    def func_custom_target(self, node: mparser.FunctionNode, args: T.Tuple[T.Optional[str]], kwargs: 'TYPE_kwargs') -> build.CustomTarget:
         if 'depfile' in kwargs and ('@BASENAME@' in kwargs['depfile'] or '@PLAINNAME@' in kwargs['depfile']):
             FeatureNew.single_use('substitutions in custom_target depfile', '0.47.0', self.subproject)
         return self._func_custom_target_impl(node, args, kwargs)
 
     def _func_custom_target_impl(self, node, args, kwargs):
         'Implementation-only, without FeatureNew checks, for internal use'
         name = args[0]
+        if name is None:
+            # name will default to first output, but we cannot do that yet because
+            # they could need substitutions (e.g. @BASENAME@) first. CustomTarget()
+            # will take care of setting a proper default but name must be an empty
+            # string in the meantime.
+            FeatureNew('custom_target() with no name argument', '0.60.0').use(self.subproject)
+            name = ''
+        elif has_path_sep(name):
+            FeatureNew('Path separator in custom_target() name argument', '0.60.0').use(self.subproject)
```

Comment:
The only difference between versions is whether or not we print a warning message about it being unsupported.

Since it works fine on old versions of meson, warning aside, I don't believe it merits a FeatureNew.

Reply:
Fair enough, removed.